components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/rabbitmq/RabbitMQUtils.java

```diff
@@ -156,9 +157,8 @@ public static void declareExchange(Connection connection, String exchangeName, H
                 if (exchangeType != null
                         && !exchangeType.equals("")) {
                     if (durable != null && !durable.equals("")) {
-                        channel.exchangeDeclare(exchangeName,
-                                exchangeType,
-                                Boolean.parseBoolean(durable));
+                        channel.exchangeDeclare(exchangeName, exchangeType, Boolean.parseBoolean(durable),
+                                Boolean.parseBoolean(autoDelete), null); // null since passing no extra arguments
                     } else {
                         channel.exchangeDeclare(exchangeName,
```

Comment:
cant there be situations where, durable is null and autoDelete = true? we are missing that right?